docs/README.md

```diff
@@ -0,0 +1,63 @@
+# Build the `cleanlab` docs **locally**
+
+1. [Fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository) and [clone](https://docs.github.com/en/get-started/quickstart/fork-a-repo#cloning-your-forked-repository) the `cleanlab` repository.
+
+2. Install the doc's required packages:
+
+```
+pip install -r docs/requirements.txt
+```
+
+3. Build the docs for all branches and tags with [`sphinx-multiversion`](https://holzhaus.github.io/sphinx-multiversion):
+
+```
+sphinx-multiversion docs/source docs/build
+```
+
+4. The docs for each branch and tag can be found in the `docs/build` directory, open the `index.html` in your browser to view the docs:
```

Comment:
won't this only build the docs for the current branch after the new updates? (i.e. a PR into master means Github Actions is only building the docs for the current master version right?)

Reply:
Yeap, that's right. The steps described here for building the docs locally and remotely are slightly different for simplicity. 

When running locally, it may be easier for users to just run the `sphinx-multiversion docs\source docs\build` command that builds the static site for all the branches and tags. But as discussed previously, this is not ideal for remote set up as it takes a long time to run and it overwrites the previously built docs.

Hence, when building it remotely with GitHub Actions, the `sphinx-multiversion` command is conditional depending on whether it is triggered by **a push to master** or **a release**:

1. If it is triggered by **a push to master**, it will run `sphinx-multiversion docs/source docs/build -D smv_branch_whitelist=master -D smv_tag_whitelist=None`. In other words, we are running the command *"build the sphinx docs with `sphinx-multiversion`, but only on the **master** branch, and **None of the tags (a superset of releases)**"*

2. If it is triggered by **a release**, it will run `sphinx-multiversion docs/source docs/build -D smv_branch_whitelist=None -D smv_tag_whitelist=v1.0.1` (or any release ID for that matter). In other words, we are running the command *"build the sphinx docs with `sphinx-multiversion`, but only on the **v1.0.1** tag, and **None of the branches**"*

Reply:
Can we have the same conditional `sphinx-multiversion` command for building the docs locally? 
I don't see why a local developer wants to wait for other branches to be built instead of just the one they are working off of.

Reply:
Fair enough, have changed the README for building the docs locally. It's longer, but it gets close to what is performed by GitHub Actions. I will add a shell script in the future.

Reply:
I still the main point is missing. Before presenting all the complicated details of what's going on behind the scenes, I just wanted the new section to say something simple like:

For the cleanlab GitHub repo, GitHub Actions workflow is configured to automatically do the following:

Whenever a commit is pushed to `master` branch, a new version of the `master` docs will be built.

Whenever a new release is tagged, a corresponding version of the docs will be built. This version will live in a new folder inside `cleanlab-docs` repo and will replace the existing `stable` version. Docs for current stable version (and all older versions) will remain as folders inside `cleanlab-docs` and will remain available as links from the sidebar.

When a user opens Pull Request, each commit to the PR will build its own version of the docs, for which a link will be posted. 
[Note: since this last functionality is not yet available in this PR, just note this last one as a TODO for the future]